examples/mobile/src/lib.rs

```diff
@@ -166,14 +166,18 @@ fn setup_music(asset_server: Res<AssetServer>, mut commands: Commands) {
 // Pause audio when app goes into background and resume when it returns.
 // This is handled by the OS on iOS, but not on Android.
 fn handle_lifetime(
-    mut lifetime_events: EventReader<ApplicationLifetime>,
+    mut lifecycle_events: EventReader<AppLifecycle>,
     music_controller: Query<&AudioSink>,
 ) {
-    for event in lifetime_events.read() {
+    let Ok(music_controller) = music_controller.get_single() else {
+        return;
+    };
+
+    for event in lifecycle_events.read() {
         match event {
-            ApplicationLifetime::Suspended => music_controller.single().pause(),
-            ApplicationLifetime::Resumed => music_controller.single().play(),
-            ApplicationLifetime::Started => (),
+            AppLifecycle::Idle | AppLifecycle::WillSuspend | AppLifecycle::WillResume => {}
```

Comment:
Should these be included in the event enum? In my testing none of these three are ever sent.

Reply:
Idle is correct (probably?) because it happens when the app just started only (and it's never reset). The other 2 though should happen but I could have missed sending them.

I have no opinions on sending Idle, it seems useless apart from initializing the state.

Reply:
Idle is never sent because lifecycle events are only sent when changed, and Idle is never changed into. WillSuspend and WillResume events are never sent because the lifecycle is always immediately changed to Suspend or Running before sending the change.

Sending idle sees pretty useless anyway when it is the default and never seen after startup.